tests/test_core/conftest.py

```diff
@@ -75,15 +16,66 @@ async def p9(request, tmpdir):
     await ctx.close()
 
 
+@pytest.fixture
+def nsname():
+    ns = uifname()
+    netns.create(ns)
+    with IPRoute(netns=ns) as ipr:
+        ipr.link('set', index=1, state='up')
+        ipr.poll(ipr.addr, 'dump', address='127.0.0.1', timeout=5)
+    yield ns
+    try:
+        netns.delete(ns)
+    except Exception:
+        pass
+
+
+@pytest.fixture
+def link(nsname):
+    ifname = uifname()
+    with IPRoute(netns=nsname) as ipr:
+        ipr.link('add', ifname=ifname, kind='dummy', state='up')
+        (link,) = ipr.poll(ipr.link, 'dump', ifname=ifname, timeout=5)
+        yield link
+        try:
+            ipr.link('del', index=link.get('index'))
+        except NetlinkError as e:
+            if e.code != errno.ENODEV:
+                raise
+
+
+@pytest.fixture
+def index(link):
+    yield link.get('index')
+
+
+@pytest.fixture
+def ifname(link):
+    yield link.get('ifname')
+
+
+@pytest.fixture
+def tmp_link(nsname):
```

Comment:
i guess there should be a sync/async version of each relevant fixture ? just a thought, i have no real idea about naming/separation yet

Reply:
Nope, there is no need to have both sync/async versions of fixtures.

The only reason why I use sync/async pair for IPRoute is because I need to test both APIs; otherwise one version is enough, and one can mix them

Reply:
since the dhcp tests i wrote are async, i had a preference for async fixtures (where the tests are async). Especially because mixing blocking sync code & async code can lead to pretty bad surprises, like:

```python
import asyncio
import time

async def its_a_trap():
    before = time.time()
    async with asyncio.timeout(1):
        time.sleep(2)
    print('took', int(time.time() - before), 'secs !')
```

(as you probably already know it takes 2 seconds.  and I have a lot of asyncio timing stuff in the client :D)

Reply:
A good point.

So far I needed only guaranteed states in the test fixtures, like "test interface created and accessible", "test netns created and set for the IPRoute" etc. Thus blocking synchronous fixtures, and the only asynchronous fixture is the one that provides AsyncIPRoute.

But that's something I have to review.

Reply:
Ok ! Understood. I think it's not needed to implement a whole set of async fixtures for now, just properly name/write fixtures so that adding async equivalents later won't be a pain